src/main/java/me/coley/recaf/mapping/AsmMappingUtils.java

```diff
@@ -0,0 +1,161 @@
+package me.coley.recaf.mapping;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Multimaps;
+
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Util class to work with ASM mappings based on String operations, such as extracting portions of ASM mapping keys or
+ * folding new mappings onto an existing mapping.
```

Comment:
Since you wrote this class feel free to add yourself to the docs with `@author`

Reply:
I personally don't think `@author` provides much value compared to what `git blame` and `git log` can do, so unless you want me to I'll pass